app/uk/gov/hmrc/nationalinsurancerecord/services/NationalInsuranceRecordService.scala

```diff
@@ -65,7 +65,7 @@ trait NpsConnection extends NationalInsuranceRecordService {
         manualCorrespondence <- manualCorrespondenceF
       ) yield {
 
-        val purgedNIRecord = desNIRecord.purge(desSummary.finalRelevantYear)
+        val purgedNIRecord = desNIRecord.purge(desSummary.finalRelevantYear.get)
```

Comment:
There are a couple of .gets in this file - better maybe to call getOrElse and either throw exception if None or default to some other value (0?)

Reply:
We can't default to other data as it wouldn't be a valid from a business angle. It is better to show a technical difficulty page than to potentially show incorrect data. If the data is missing it would be a bigger issue and we would have to resolve it on the other end. None.get would be fine to bring the error page to me.

Reply:
OK fair enough